waku/v2/protocol/envelope.go

```diff
@@ -1,27 +1,38 @@
 package protocol
 
-import "github.com/status-im/go-waku/waku/v2/protocol/pb"
+import (
+	"crypto/sha256"
+
+	"github.com/status-im/go-waku/waku/v2/protocol/pb"
+)
 
 // Envelope contains information about the pubsub topic of a WakuMessage
 // and a hash used to identify a message based on the bytes of a WakuMessage
 // protobuffer
 type Envelope struct {
-	msg         *pb.WakuMessage
-	pubsubTopic string
-	size        int
-	hash        []byte
+	msg   *pb.WakuMessage
+	size  int
+	hash  []byte
+	index *pb.Index
 }
 
 // NewEnvelope creates a new Envelope that contains a WakuMessage
 // It's used as a way to know to which Pubsub topic belongs a WakuMessage
 // as well as generating a hash based on the bytes that compose the message
-func NewEnvelope(msg *pb.WakuMessage, pubSubTopic string) *Envelope {
+func NewEnvelope(msg *pb.WakuMessage, receiverTime int64, pubSubTopic string) *Envelope {
 	data, _ := msg.Marshal()
+	hash := sha256.Sum256(append([]byte(msg.ContentTopic), msg.Payload...))
+
 	return &Envelope{
-		msg:         msg,
-		pubsubTopic: pubSubTopic,
-		size:        len(data),
-		hash:        pb.Hash(data),
+		msg:  msg,
+		size: len(data),
+		hash: pb.Hash(data),
+		index: &pb.Index{
```

Comment:
I have a feeling this interface change is going to require some updates to our `store` implementation. Should be pretty minor.